server/src/main/java/org/elasticsearch/monitor/jvm/SunThreadInfo.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.monitor.jvm;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.lang.management.ThreadMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.reflect.Method;
+
+public class SunThreadInfo {
+
+    private static final ThreadMXBean threadMXBean;
+    private static final Method getThreadAllocatedBytes;
+    private static final Method isThreadAllocatedMemorySupported;
+    private static final Method isThreadAllocatedMemoryEnabled;
+
+    private static final Logger logger = LogManager.getLogger(SunThreadInfo.class);
+
+    static {
+        threadMXBean = ManagementFactory.getThreadMXBean();
+        getThreadAllocatedBytes = getMethod("getThreadAllocatedBytes", long.class);
+        isThreadAllocatedMemorySupported = getMethod("isThreadAllocatedMemorySupported");
+        isThreadAllocatedMemoryEnabled = getMethod("isThreadAllocatedMemoryEnabled");
+    }
+
+    public static boolean isThreadAllocatedMemorySupported() {
+        if (isThreadAllocatedMemorySupported == null) {
+            logger.warn("isThreadAllocatedMemorySupported is not available");
```

Comment:
I think it's best if we remove the logger.warn messages. I just realized that `getThreadAllocatedBytes` is called for every thread in any report mode, which means that on a JVM that doesn't support his API we'll log these warnings in the Elasticsearch logs for as many threads as we have running. It's probably not the best.

Reply:
Yeah, It is really a issue, i will remove the log message, if `getThreadAllocatedBytes`  is not support, the exception message seems enough: 

`ElasticsearchException("thread allocated memory is not supported on this JDK");`

I  also add a judgement, only call `getThreadAllocatedBytes` in `mem` mode, what do you think of this?